x-pack/plugins/fleet/server/collectors/integrations_collector.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsClient } from '@kbn/core/server';
+
+import { SO_SEARCH_LIMIT } from '../constants';
+import { packagePolicyService } from '../services';
+
+export interface IntegrationsDetails {
+  totalPoliciesCount: number;
+  totalSharedCount: number;
+  sharedIntegrations: SharedIntegration[];
+}
+
+interface SharedIntegration {
+  name: string;
+  sharedByPoliciesCount: number;
+  pkgName?: string;
+  pkgVersion?: string;
+  agents?: number;
+}
+
+export const getIntegrationsDetails = async (
+  soClient?: SavedObjectsClient
+): Promise<IntegrationsDetails | undefined> => {
+  if (!soClient) {
+    return undefined;
+  }
+  const allPackagePolicies = await packagePolicyService.list(soClient, {
```

Comment:
Couldn't we query the `policy_ids.length > 1` directly in ES?

Reply:
The only concern here is if there are many package policies in a cluster, we would read them all at once in the memory.  I'm not sure if we have an official limit of how many package policies are supported, only 500 agent policies. We might want to revisit later to optimize.

Reply:
At the moment there is the `SO_SEARCH_LIMIT` set on the query, we could also lower that or optimize later.